addons/web_editor/static/src/js/common/grid_layout_utils.js

```diff
@@ -280,7 +280,8 @@ export function _convertColumnToGrid(rowEl, columnEl, columnWidth, columnHeight)
 
     // Computing the column and row spans.
     const gridProp = _getGridProperties(rowEl);
-    const columnColCount = Math.round((columnWidth + gridProp.columnGap) / (gridProp.columnSize + gridProp.columnGap));
+    // As the number of column can not exceed 12.
+    const columnColCount = Math.min(12, Math.round((columnWidth + gridProp.columnGap) / (gridProp.columnSize + gridProp.columnGap)));
```

Comment:
The biggest problem of this is that for example, the "Chart" snippet and the "Donation Button" are both `div`. However, the size of the "Chart" is much bigger than the "Donation Button" so if you set the size of the drag helper to be the same for those two, it will be quite big for the "Donation button" which will may be do not look really good ;)

Reply:
Okay indeed you are right, it might become too overkill to manage the particular cases :slightly_smiling_face: 

However, I still think that some snippets have a width that is too big (e.g. alert, rating,... 12 columns is too much for them) :thinking: Since it is alright to modify the templates and since the attributes won't stay after the drop, why don't you also add starting heights and widths for the other inner contents, instead of just those with dynamic content ? :slightly_smiling_face: Like this, they will all look better when dropped :wink: 
Of course it's just my opinion, maybe ask PDE what he thinks of that :slightly_smiling_face:

Reply:
@sobo-odoo : FYI: pde confirmed that he prefers to have not too big inner snippets coming from the right panel; if a snippet comes from the right panel, its width should not be bigger than 6 columns in the grid. To do so, I preferred to adapt the js rather than modifying the xml of those snippets. 

I preferred to do like this for those reasons:
- I preferred to reserved the xml solution where the dimension have to be wrote down "manually" for the exception of the dynamically built snippet (and few others snippets when there is no other choice). Indeed, I like that fact that there is a general solution that works for most of the snippets and that will work if a new inner snippet is developed. In this case, the developer in charge of the creation of the new snippet will not need to write the snippet dimension in the xml for the feature of this PR to work (Note: this is only true if the new snippet is not built dynamically).
- It is in consistency with how it works for the drag of the `SnippetEditor`.
- As the snippet dimension can be computed, it would be a "redundant" information to put in the xml (even if this is not added on the client website page).

However, do not hesitate to give me your opinion about this :)

Reply:
@loco-odoo Thanks for telling me ! :slightly_smiling_face: I agree with your arguments :wink: And I like your solution because it did not require to modify the function in `grid_layout_utils.js` so it keeps it simple to understand :+1: